scripts/latent_alignment_reranker.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import gzip
+import os
+
+from allennlp.common import Params
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.dataset_readers import LatentAlignmentDatasetReader
+from allennlp.data.dataset_readers import WikiTablesDatasetReader
+from allennlp.models.archival import load_archive
+from allennlp.data.dataset_readers.semantic_parsing.wikitables import util 
+from tqdm import tqdm
+
+def rerank_lf(model_file, input_examples_file, params_file, tables_directory, lf_directory, output_directory):
+    model = load_archive(model_file).model
+    model.eval()
+
+
+    params = Params.from_file(params_file)
+    latent_alignment_reader = DatasetReader.from_params(params.pop('dataset_reader'))
+
+    with open(input_examples_file) as input_file:
+        input_lines = input_file.readlines()
+
+    found = 0.0
+    for line in tqdm(input_lines):
+        parsed_info = util.parse_example_line(line)
+        example_id = parsed_info["id"]
+        lf_output_filename = os.path.join(lf_directory, parsed_info["id"] + '.gz')
+        try:
+            lf_file = gzip.open(lf_output_filename)
+            sempre_forms = [lf.strip().decode('utf-8') for lf in lf_file]
+            question = parsed_info['question']
+            instance = latent_alignment_reader.text_to_instance(question, sempre_forms)
+            output = model.forward_on_instance(instance)
+            similarities = output['all_similarities'] 
+            top_lfs = [lf for lf, score in sorted(zip(sempre_forms, similarities), key = lambda x: x[1], reverse = True)[:10]]
+            if not os.path.exists(output_directory):
+                os.makedirs(output_directory)
+            output_file = gzip.open(os.path.join(output_directory, f"{example_id}.gz"), "wb")
```

Comment:
Minor comment: You can use the mode `wt`, and not worry about the `utf-8` encoding below.